src/imageproxy.rs

```diff
@@ -525,16 +565,16 @@ impl ImageProxy {
     #[instrument]
     pub async fn close_image(&self, img: &OpenedImage) -> Result<()> {
         tracing::debug!("closing image");
-        let (r, [], []) = self.impl_request("CloseImage", [img.0]).await?;
+        let (r, ()) = self.impl_request("CloseImage", [img.0]).await?;
         Ok(r)
     }
 
-    async fn read_all_fd(&self, datafd: OwnedFd, pipeid: PipeId) -> Result<Vec<u8>> {
-        let fd = tokio::fs::File::from_std(std::fs::File::from(datafd));
+    async fn read_finish_pipe(&self, pipe: FinishPipe) -> Result<Vec<u8>> {
```

Comment:
Specifically for example I think this is now clearer as we have a struct that holds the state of these two all the way to the end.

Reply:
Ya.   I was thinking the same and was a bit sad to lose it...